src-django/api/field.py

```diff
@@ -14,3 +15,56 @@ def to_internal_value(self, data):
             raise serializers.ValidationError('All choices should be strings')
 
         return json.dumps(data)
+
+
+class ConditionTreeField(serializers.Field):
+    def to_representation(self, obj):
+        return json.loads(obj)
+
+    def to_internal_value(self, data):
+        if not isinstance(data, dict):
+            raise serializers.ValidationError('Expected dict')
+
+        self.__validate_json(data)
+
+        return json.dumps(data)
+
+    def __validate_json(self, data):
+        if 'node_type' not in data or not isinstance(data['node_type'], basestring):
+            raise serializers.ValidationError('Missing or invalid node type')
+
+        if data['node_type'] in ShowIf.CRITERIA_TYPES:
+            if 'children' in data:
+                raise serializers.ValidationError('CRITERIA node must have no children')
+
+            if 'criteria_element' not in data or not isinstance(data['criteria_element'], int):
+                raise serializers.ValidationError('CRITERIA must have an element')
+
+            try:
```

Comment:
This check should only be in the generator. A user should be allowed to create an empty criteria node on the front end